Added access logs topic (#244)
agentgateway/website: assets/agw-docs/pages/security/access-logging.md

* The downstream (client) address, connection information, TLS configuration, and timing
* The backend (service) address, connection information, TLS configuration, timing, and routing information
* Relevant configuration, such as rate of sampling (if used)
* Filter-specific context that is published to the dynamic metadata during the filter chain

### Additional HTTP properties 

When the gateway is used as an HTTP proxy, additional HTTP information is available for access logging, including:

* Request data, including the method, path, scheme, port, user agent, headers, body, and more
* Response data, including the response code, headers, body, and trailers, as well as a string representation of the response code
* Protocol version

{{< reuse "agw-docs/snippets/agentgateway/prereq.md" >}}

## Access logs to `stdout` {#access-log-stdout-filesink}

You can set up access logs to write to a standard (stdout/stderr) stream. The following example writes access logs to a stdout in the pod of the selected `agentgateway-proxy` gateway.

> 1. Create an {{< reuse "agw-docs/snippets/trafficpolicy.md" >}} resource to define your access logging rules. The following example writes access logs to the `stdout` stream of the gateway proxy container when a request fails. It also adds a string field of the response code. Successful requests are not logged with this configuration.

Reviewer: ```suggestion
1. Create an {{< reuse "agw-docs/snippets/trafficpolicy.md" >}} resource to define your access logging rules. The following example writes access logs to the `stdout` stream of the gateway proxy container when a request fails with a 404 HTTP response code. It also adds the actual response code to the log entry. This policy does nto apply to requests that return a response code other than 404.
```

{{< reuse "agw-docs/snippets/agentgateway/prereq.md" >}}

## Access logs to `stdout` {#access-log-stdout-filesink}

You can set up access logs to write to a standard (stdout/stderr) stream. The following example writes access logs to a stdout in the pod of the selected `agentgateway-proxy` gateway.

1. Create an {{< reuse "agw-docs/snippets/trafficpolicy.md" >}} resource to define your access logging rules. The following example writes access logs to the `stdout` stream of the gateway proxy container when a request fails with a 404 HTTP response code. It also adds the actual response code to the log entry. This policy does not apply to requests that return a response code other than 404.

   ```yaml {linenos=table,paths="access-logging"}
   kubectl apply -f- <<EOF
   apiVersion: agentgateway.dev/v1alpha1
   kind: AgentgatewayPolicy
   metadata:
     name: access-logs

>      namespace: agentgateway-system

Reviewer: ```suggestion
     namespace: {{< reuse "agw-docs/snippets/namespace.md" >}}
```

   spec:
     targetRefs:
     - group: gateway.networking.k8s.io
       kind: Gateway
       name: agentgateway-proxy
     frontend:
       accessLog:
         filter: response.code == 404
         attributes:
           add:
           - name: http.statusString
             expression: string(response.code)
   EOF
   ```

   | Setting | Description |
   | ------- | ----------- |
   | `targetRefs`| Select the Gateway to enable access logging for. The example selects the `agentgateway-proxy` gateway that you created from the sample app guide. |
   | `accessLog` | Configure the details for access logging. |
   | `filter` | Filter the logs that are included by using a CEL expression. |

>    | `attributes` | Add or remove attributes that are logged in the requests by using a CEL expression. |

Reviewer: ```suggestion
   | `attributes` | Add or remove attributes from the access log by using a CEL expression. |
```
